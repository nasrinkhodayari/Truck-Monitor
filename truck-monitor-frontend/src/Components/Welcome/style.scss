@import "../../Assets/scss/variables.scss";
.welcome-container {
    z-index: 999;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: $light_blue_gray;
    .welcome {
        width: 50%;
        margin: 0 auto;
        top: 25%;
        z-index: 1000;
        @media screen and (max-width: 992px) {
           width: 100%;
           top: 0;
        }
    }
}
