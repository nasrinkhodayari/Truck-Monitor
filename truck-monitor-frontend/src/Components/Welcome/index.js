import React, { useState } from "react";
import Alert from 'react-bootstrap/Alert';
import Button from 'react-bootstrap/Button';
import BootstrapSwitchButton from 'bootstrap-switch-button-react';
import './style.scss';
import I18nInstance from '../../Utils/i18n';

const Welcome = props => {
    const { showWelcomeHelp,setShowWelcomeHelp } = props;
    const [languageEnglish, setLanguageEnglish] = useState(I18nInstance.language === "en" ? true : false);
    return (
        showWelcomeHelp && <div className="welcome-container">
            <Alert variant="success" className="welcome">
                <Alert.Heading>Hi, nice to see you</Alert.Heading>
                <p>This is a sample for <strong>MAN Truck Monitor</strong>,there is a truck
                in my database for presenting system functionality to you,
                firstly you need to enter the truck license plate ,
                so please use this license plate number :<strong> FN67KE</strong> ,
                if you enter wrong licesne plate number,system will warn you.
                Also this application support two language (pt/en) it set base on your browser language ,
                 but you can change it using this switch button.
                 </p>
                Change Language: <BootstrapSwitchButton
                        checked={languageEnglish}
                        onlabel='en'
                        offlabel='pt'
                        size="xs"
                        onChange={checked => {
                            setLanguageEnglish(checked);
                            I18nInstance.changeLanguage(checked ? "en" : "pt");
                            localStorage.setItem('lang', I18nInstance.language);
                        }}
                    />
                <hr />
                <p><strong>Thank you in advance for your attention :)</strong></p>
                <div className="d-flex justify-content-end">
                    <Button onClick={() => setShowWelcomeHelp(false)} variant="outline-success">Got it</Button>
                </div>
            </Alert>
        </div>
    );
}
export default Welcome;