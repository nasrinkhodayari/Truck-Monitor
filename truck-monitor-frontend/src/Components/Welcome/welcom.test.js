import React from 'react';
import { shallow } from 'enzyme';
import Welcome from './index';

describe('App Welcome component', () => {
    it('Welcome component renders correctlly', () => {
        const wrapper = shallow(<Welcome />);
        expect(wrapper).toMatchSnapshot();
    });
});