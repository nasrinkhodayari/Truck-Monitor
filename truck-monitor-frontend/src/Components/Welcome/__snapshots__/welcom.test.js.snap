// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Welcome component Welcome component renders correctlly 1`] = `ShallowWrapper {}`;
